### PR TITLE
Use random_bytes() when appropriate.

### PR DESCRIPTION
**Pull request by cweagans**

### CODE REVIEW

File: phpseclib/Crypt/Random.php [hunk 1]
```diff
@@ -53,6 +53,20 @@ class Random
      */
     public static function string($length)
     {
+        if (version_compare(PHP_VERSION, '7.0.0', '>=')) {
+            try {
+                $string = random_bytes($length);
```
Comment (GrahamCampbell) [inline phpseclib/Crypt/Random.php]: don't need the variable asignment

File: phpseclib/Crypt/Random.php [hunk 2]
```diff
@@ -53,6 +53,20 @@ class Random
...
+                return $string;
+            }
+            catch (\EngineException $e) {
```
Comment (GrahamCampbell) [inline phpseclib/Crypt/Random.php]: there is no such class as EngineException - it was renamed to "Error"

Comment (GrahamCampbell) [inline phpseclib/Crypt/Random.php]: see https://3v4l.org/jfh8F

File: phpseclib/Crypt/Random.php [hunk 3]
```diff
@@ -53,6 +53,19 @@ class Random
      */
     public static function string($length)
     {
+        if (version_compare(PHP_VERSION, '7.0.0', '>=')) {
+            try {
+                return random_bytes($length);
+            }
```
Comment (bantu) [inline phpseclib/Crypt/Random.php]: The catch statement need to go here I think

Comment (cweagans) [inline phpseclib/Crypt/Random.php]: Fixed.

**Comment by bantu** [comment]
This is no longer required after #794

**Comment by cweagans** [comment]
#794 is against a different branch.

**Comment by bantu** [comment]
@cweagans We merge 1.0 into 2.0 and 2.0 into master.

**Comment by cweagans** [comment]
Ah, I see. :+1:

**Pull request closed without merging**